streams/src/main/java/org/apache/kafka/streams/processor/internals/GlobalStreamThread.java

```diff
@@ -310,6 +317,25 @@ public void run() {
                     cache.resize(size);
                 }
                 stateConsumer.pollAndUpdate();
+
+                if (fetchDeadline != -1) {
+                    if (fetchDeadline > time.milliseconds()) {
+                        try {
+                            globalConsumerClientInstanceId = globalConsumer.clientInstanceId(Duration.ZERO);
```

Comment:
That's a bit impatient.

Reply:
Yes, but this was intentionally. The `GlobalStreamThread` does this call "on the side", and thus the idea is to just call it with no timeout to just trigger the background RPC and not block the thread from doing its actually work at all. -- There won't be a busy wait, because the global thread will do other useful work in the meantime before it cycles back here, to check if the RPC returned.

Thoughts?